wicket/src/screens/update.rs

```diff
@@ -0,0 +1,192 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! Manage updates for rack components
+
+use super::common::CommonScreenState;
+use super::{Screen, ScreenId};
+use crate::defaults::style;
+use crate::widgets::{HelpButtonState, HelpMenuState};
+use crate::wizard::Frame;
+use crate::wizard::{Action, ScreenEvent, State, Term};
+use crate::{BOTTOM_MARGIN, TOP_MARGIN};
+use crossterm::event::Event as TermEvent;
+use crossterm::event::{
+    KeyCode, KeyEvent, MouseButton, MouseEvent, MouseEventKind,
+};
+use tui::layout::Alignment;
+use tui::layout::Rect;
+use tui::text::Spans;
+use tui::widgets::Block;
+
+pub struct UpdateScreen {
+    common: CommonScreenState,
+}
+
+impl UpdateScreen {
+    pub fn new() -> UpdateScreen {
+        let help_text = vec![
+            ("<ARROWS> | Mouse scroll", "Cycle through components"),
+            ("<CTRL-h>", "Toggle this help menu"),
+            ("<CTRL-n>", "Goto the next screen"),
+            ("<CTRL-p>", "Goto the previous screen"),
+            ("<CTRL-c>", "Exit the program"),
+        ];
+
+        UpdateScreen {
+            common: CommonScreenState {
+                hovered: None,
+                help_button_state: HelpButtonState::new(1, 0),
+                help_menu_state: HelpMenuState::new(help_text),
+                prev_screen: Some(ScreenId::Component),
+                next_screen: ScreenId::Rack,
+            },
+        }
+    }
+
+    fn handle_key_event(
+        &mut self,
+        state: &mut State,
+        event: KeyEvent,
+    ) -> Vec<Action> {
+        match event.code {
+            KeyCode::Tab => {
```

Comment:
Currently, `rack_state` contains a `TabIndex` to cycle through components. We maintain this state in both the `Rack` and `Components` screen, meaning when you switch between them the same component is selected. This isn't currently maintained across the update screen, and I'm not sure it makes sense there. Currently there is actually no mapping from the list to the rack state `TabIndex`. We should add that support if it makes sense, or remove the tab controls here. We can also allow tab controls just for scroll up/scroll down, which we probably should do for symmetry.

Reply:
Couple comments:

* The tab/backtab on the update screen _appear_ to not do anything at all, until you go back to the previous screen. I'd vote for changing tab controls to scroll up/down on this screen and not affect the previous screen.
* If I switch between the inventory and update screens, each remembers the last sled selected but they're not related to each other; it feels weird. Maybe updates shouldn't be an "after" screen from inventory, but both are reachable from the rack screen? Connecting the selected sled in the update screen with the selected sled on the rack screen seems more reasonable if there's not the inventory screen in between. (This feels like a bigger change - even if you like it I'd defer it and just leave the two screens selections' unconnected for now.)

Reply:
Yeah, I agree this isn't right as is. I'm going to discuss this with @benjaminleonard. As mentioned offline, I think it may be better to just move the inventory into the update screen as a separate tab with the list view for scrolling between both update history and inventory.

Reply:
I had a good chat with @benjaminleonard. We're likely going restructure the pages altogether. For know I'm going to make tabbing go up/down the list.

Reply:
Fixed.